src/java/org/apache/cassandra/service/AutoRepairServiceMBean.java

```diff
@@ -71,4 +70,8 @@ public interface AutoRepairServiceMBean
     public void setRepairSessionTimeout(RepairType repairType, String timeout);
 
     public Set<String> getOnGoingRepairHostIds(RepairType rType);
+
+    public Map<String, String> getAutoRepairTokenRangeSplitterParameters(RepairType repairType);
```

Comment:
I don't think we can expose non-core JDK types through mBeans, we try to expose only native types . I see issues with both `RepairType` and `InetAddressAndPort` in this class. Especially concerning for `InetAddressAndPort`, we have methods to stringify the value see `org.apache.cassandra.service.StorageService#getNaturalEndpointsWithPort(java.lang.String, java.lang.String, java.lang.String)`. Maybe exposing an enum is fine though, so `RepairType` might be okay, but I have not seen this being done anywhere else in the beans that we expose through JMX

Reply:
Ah great point.  Given `RepairType` is a simple enum it might be ok, but I agree that exposing `AutoRepairConfig` and `InetAddressAndPort`d can be a problem, especially if we ever evolve the type.  I think I agree we should try to change this to only expose java types , especially since I don't see precedence for doing this anywhere else in C*.

@jaydeepkumar1984 thoughts?  Maybe we can attack this in another jira.